### PR TITLE
New design for media collections

### PR DESCRIPTION
**Pull request by marcelmoosbrugger**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | yes
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | fixes https://github.com/sulu/sulu/issues/2577
| Related issues/PRs | https://github.com/sulu/sulu/pull/2765, https://github.com/massiveart/husky/pull/706, https://github.com/massiveart/husky/pull/712
| License | MIT
| Documentation PR | none

#### What's in this PR?

This PR implements the new design for the media collections.

### CODE REVIEW

**Comment by marcelmoosbrugger** [comment]
@danrot This PR is ready for review.

**Comment by marcelmoosbrugger** [comment]
@danrot ping.

File: src/Sulu/Bundle/MediaBundle/Resources/public/js/components/collections/edit/files/main.js [hunk 1]
```diff
@@ -94,6 +106,9 @@ define([
             // add uploaded medias to datagrid
             this.sandbox.on('husky.dropzone.' + this.options.instanceName + '.success', function(file, mediaResponse) {
                 this.sandbox.emit('sulu.labels.success.show', 'labels.success.media-upload-desc', 'labels.success');
+                mediaResponse = this.sandbox.util.extend(true, mediaResponse, {
```
Comment (danrot) [inline src/Sulu/Bundle/MediaBundle/Resources/public/js/components/collections/edit/files/main.js]: Why has it to be cloned? And I would prefer to use the `deepCopy` method for that.

File: src/Sulu/Bundle/MediaBundle/Resources/public/js/main.js [hunk 2]
```diff
@@ -56,28 +57,45 @@ define([
...
+                callback: function(locale, id) {
+                    return '<div data-aura-component="collections/edit@sulumedia" data-aura-id="' + id + '" data-aura-locale="' + locale + '"/>';
                 }
             });
 
             // show a single collection with files and upload
             sandbox.mvc.routes.push({
+                route: 'media/collections/:locale/edit::id/:content/edit::mediaId',
+                callback: function(locale, id, content, mediaId) {
+                    sandbox.sulu.viewStates['media-file-edit-id'] = parseInt(mediaId);
```
Comment (danrot) [inline src/Sulu/Bundle/MediaBundle/Resources/public/js/main.js]: From a pure feeling this is not really correct here... Shouldn't this be handled by the component somehow? This routing file is supposed to be as simple as possible.

**Pull request merged**